hoss/coordinate_utilities.py

```diff
@@ -162,59 +165,287 @@ def get_coordinate_array(
+        lat_arr,
+        lon_arr,
+    )
+
+    row_indices, col_indices = get_valid_row_col_pairs(
+        lat_arr, lon_arr, latitude_coordinate, longitude_coordinate
+    )
+
+    y_x_order, projected_y = get_dimension_order(
+        lat_arr, lon_arr, row_indices, True, projected_dimension_names
+    )
+    # if is row, the index_for_dimension is 0
+    y_dim = get_dimension_array_from_geo_points(
+        lat_arr, lon_arr, crs, row_indices, row_size, 0, y_x_order
+    )
+
+    x_y_order, projected_x = get_dimension_order(
+        lat_arr,
+        lon_arr,
+        col_indices,
+        False,
+        projected_dimension_names,
+    )
+    # if is column, the index_for_dimension is 1
+    x_dim = get_dimension_array_from_geo_points(
+        lat_arr, lon_arr, crs, col_indices, col_size, 1, x_y_order
+    )
+
+    # projected_y, projected_x = tuple(projected_dimension_names)
+    if y_x_order == 1:
+        return {projected_y: y_dim, projected_x: x_dim}
+    return {projected_x: x_dim, projected_y: y_dim}
+
+
+def get_dimension_array_from_geo_points(
+    lat_arr: np.ndarray,
+    lon_arr: np.ndarray,
+    crs: CRS,
+    dimension_indices: list,
+    dimension_size: int,
+    index_for_dimension: int,
+    x_or_y_index: int,
+) -> np.ndarray:
+    """This function uses the list of lat/lon points corresponding
+    to a list of array indices and reprojects it with the CRS
+    provided and scales the x/y values to a dimension array with the dimension
+    size provided. index_for_dimension is 0 for row and
+    1 for column. x_or_y_index is 1 if latitude is used for row
+    and is 0 if longitude is used for row. (it determines the dimension order)
+
+    """
+
+    geo_grid_points = [
+        (lon_arr[row, col], lat_arr[row, col]) for row, col in dimension_indices
+    ]
+
+    x_y_values = get_x_y_values_from_geographic_points(geo_grid_points, crs)
+
+    indices_for_dimension = np.transpose(dimension_indices)[index_for_dimension]
+
+    dimension_values = np.transpose(x_y_values)[x_or_y_index]
+
+    return get_1d_dim_array_data_from_dimvalues(
+        dimension_values, indices_for_dimension, dimension_size
+    )
+
+
+def get_dimension_order(
+    lat_array_points: np.ndarray,
+    lon_array_points: np.ndarray,
+    dimension_indices: list,
+    is_row: bool,
+    projected_dimension_names: list,
+) -> list[int, str]:
+    """Determines the order of dimensions based on whether the
+    latitude and longitude are varying across row or column
+    """
+    projected_y, projected_x = tuple(projected_dimension_names)
+
+    # if lat/lon array is 2D and variables are also 2D
+    lat_arr_values = [lat_array_points[i][j] for i, j in dimension_indices]
+    lon_arr_values = [lon_array_points[i][j] for i, j in dimension_indices]
+
+    # if it is row and lat is changing return 1
+    # and call it y
+    if is_row is True and np.all(np.diff(lat_arr_values) != 0):
+        return (1, projected_y)
+
```

Comment:
I wonder if we should be using not is_close for floating point, rather than an absolute != 0? (and in following statements)

Reply:
np.isclose is for comparing two arrays I think. will check into it